pytorch_tcn/tcn.py

```diff
@@ -383,6 +388,10 @@ def __init__(
         self.use_gate = use_gate
 
         if embedding_shapes is not None:
+            if not isinstance(embedding_shapes, list):
+                # Allow for yaml config file instantiation
+                embedding_shapes = [tuple( shape ) for shape in embedding_shapes]
+
             if isinstance(embedding_shapes, list):
```

Comment:
It may be cleaner to check for an iterable here, like this:

```python
from collections.abc import Iterable
...
if isinstance( embedding_shapes, Iterable ):
    for shape in embedding_shapes:
        if not isinstance( shape, tuple ):
            # try to make shape a tuple and catch exception 
            # and if not possible, throw respective error
```

Would this work with the hydra instantiation?